packages/flutter_tools/lib/src/commands/channel.dart

```diff
@@ -24,18 +34,20 @@ class ChannelCommand extends FlutterCommand {
   Future<Null> runCommand() {
     switch (argResults.rest.length) {
       case 0:
-        return _listChannels();
+        return listChannels(showAll: argResults['all']);
       case 1:
-        return _switchChannel(argResults.rest[0]);
+        return switchChannel(argResults.rest[0]);
       default:
         throw new ToolExit('Too many arguments.\n$usage');
     }
   }
 
-  Future<Null> _listChannels() async {
-    final String currentBranch = runSync(
-        <String>['git', 'rev-parse', '--abbrev-ref', 'HEAD'],
-        workingDirectory: Cache.flutterRoot);
+  static Future<Null> listChannels({ bool showAll }) async {
```

Comment:
Why public?  And if it were still private, it could be an instance method that just refers to `argResults` directly.

Reply:
Made it private again. I had thought I was going to use this from other files, but only upgradeChannel ended up being used.

I left it using an argument rather than argResults.